vendor/bat-native-ads/src/bat/ads/internal/database/tables/creative_new_tab_page_ads_database_table.h

```diff
@@ -0,0 +1,117 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BAT_ADS_INTERNAL_DATABASE_CREATIVE_NEW_TAB_PAGE_ADS_DATABASE_TABLE_H_
+#define BAT_ADS_INTERNAL_DATABASE_CREATIVE_NEW_TAB_PAGE_ADS_DATABASE_TABLE_H_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "bat/ads/ads_client.h"
+#include "bat/ads/internal/bundle/creative_new_tab_page_ad_info.h"
+#include "bat/ads/internal/classification/page_classifier/page_classifier.h"
+#include "bat/ads/internal/database/database_table.h"
+#include "bat/ads/internal/database/tables/campaigns_database_table.h"
+#include "bat/ads/internal/database/tables/categories_database_table.h"
+#include "bat/ads/internal/database/tables/creative_ads_database_table.h"
+#include "bat/ads/internal/database/tables/geo_targets_database_table.h"
+#include "bat/ads/mojom.h"
+#include "bat/ads/result.h"
+
+namespace ads {
+
+using GetCreativeNewTabPageAdCallback = std::function<void(const Result,
+    const CreativeNewTabPageAdInfo&)>;
+
+using GetCreativeNewTabPageAdsCallback = std::function<void(const Result,
+    const std::vector<std::string>&, const CreativeNewTabPageAdList&)>;
+
+class AdsImpl;
+
+namespace database {
+namespace table {
+
+class CreativeNewTabPageAds : public Table {
+ public:
+  explicit CreativeNewTabPageAds(
+      AdsImpl* ads);
+
+  ~CreativeNewTabPageAds() override;
+
+  void Save(
+      const CreativeNewTabPageAdList& creative_new_tab_page_ads,
+      ResultCallback callback);
+
+  void GetCreativeNewTabPageAd(
+      const std::string& creative_instance_id,
```

Comment:
Maybe this is a big refactor that should be handled in a separate PR, but it's throwing me off that we have shared ID's without corresponding tables for `creative_instance_id`, and`creative_set_id`. Additionally, I think it makes more sense to call this `creative_id`, as `instance` is implied.

Reply:
Discussed and agreed to rename `creative_ads` table to `creative_sets`